### PR TITLE
diagnostics: all pprofs

### PR DESCRIPTION
**Pull request by dvovk**
Extended pprof read API to include:  goroutine, threadcreate, heap, allocs, block, mutex

### CODE REVIEW

File: diagnostics/profile.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+func SetupProfileAccess(metricsMux *http.ServeMux, diag *diaglib.DiagnosticClient) {
+	if metricsMux == nil {
+		return
+	}
+
+	//handle all pprof, supported: goroutine, threadcreate, heap, allocs, block, mutex
+	metricsMux.HandleFunc("/pprof/", func(w http.ResponseWriter, r *http.Request) {
+		profile := r.URL.Path[len("/pprof/"):]
```
Comment (shohamc1) [inline diagnostics/profile.go:34]: You can use [strings.TrimPrefix](https://pkg.go.dev/strings#TrimPrefix). Alternatively, you can also use HTTP params using `r.URL.Query().Get("profile")`, but this changes your request format to `/pprof?profile=<name>`.

File: diagnostics/profile.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		writePproProfile(w, profile)
+	})
+}
+
+func writePproProfile(w http.ResponseWriter, profile string) {
```
Comment (shohamc1) [inline diagnostics/profile.go:39]:
```suggestion
func writeProfProfile(w http.ResponseWriter, profile string) {
```

File: diagnostics/profile.go (hunk 1 above)
Comment (dvovk) [reply diagnostics/profile.go]: done

File: diagnostics/profile.go (hunk 2 above)
Comment (dvovk) [reply diagnostics/profile.go]: fixed typo, I want it to have "Pprof" to point that I and getting data from this lib

**Review by taratorio — APPROVED** [review]

**Pull request merged**